cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/sidecar/ArtifactLocalizerService.java

```diff
@@ -51,6 +63,95 @@
       .setHttpHandlers(new ArtifactLocalizerHttpHandlerInternal(artifactLocalizer));
 
     httpService = builder.build();
+    cacheDir = Paths.get(cConf.get(Constants.CFG_LOCAL_DATA_DIR));
+    cacheCleanupInterval = cConf.getInt(Constants.ArtifactLocalizer.CACHE_CLEANUP_INTERVAL_MIN);
+  }
+
+  @Override
+  @VisibleForTesting
+  protected void runOneIteration() throws Exception {
+    try {
+      cleanupArtifactCache(cacheDir.resolve("artifacts").toFile());
+    } catch (Exception e) {
+      LOG.warn("ArtifactLocalizerService failed to clean up cache.", e);
```

Comment:
If failure happens, the job will be retried right? In that case probably we should mention that in the log.

Reply:
done